main/src/addins/MacPlatform/MacInterop/Carbon.cs

```
@@ -215,15 +215,15 @@ internal static string UnConvertCharCode (int i)
 		
 		[DllImport (CarbonLib)]
 		static extern int CPSSetProcessName (ref ProcessSerialNumber psn, string name);
-		
+
 		public static void SetProcessName (string name)
 		{
 			try {
 				if (GetCurrentProcess (out ProcessSerialNumber psn) == 0)
 					CPSSetProcessName (ref psn, name);
-			} catch {} //EntryPointNotFoundException?
+			} catch (TypeLoadException) {} //EntryPointNotFoundException?
 		}
```

Previously we were catching all exceptions now we are handling only a TypeLoadException. Is that the only exception that is thrown here? Do we not want to catch all exceptions here? If this fails with another exception do we want it to stop/break the IDE?

---

Well, given the only 2 methods being called here are PInvokes, those are kinda the only exceptions that can be thrown.

---

There are 2 other possible pinvokes exceptions (ExecutionEngine and AccessViolation), but both tear down the runtime.

---

Was just wondering 👍